klippy/extras/homing_heaters.py

```diff
@@ -0,0 +1,54 @@
+# Heater handling on homing moves
+#
+# Copyright (C) 2016-2018  Kevin O'Connor <kevin@koconnor.net>
+#
+# This file may be distributed under the terms of the GNU GPLv3 license.
+
+import logging
+
+class HomingHeaters:
+    def __init__(self, config):
+        self.printer = config.get_printer()
+        self.printer.register_event_handler("homing:move_begin",
+                                            self.handle_homing_move_begin)
+        self.printer.register_event_handler("homing:move_end",
+                                            self.handle_homing_move_end)
+        self.heaters_to_disable = config.get("heaters", "")
+        self.steppers_needing_quiet = config.get("steppers", "")
+        self.last_target = {}
+        self.pheater = self.printer.lookup_object('heater')
+
+    def check_eligible(self, endstops):
+        flaky_steppers = [n.strip()
+                         for n in self.steppers_needing_quiet.split(',')]
+        if flaky_steppers == [""]:
+            return True
+        endstop_instances, dummy = zip(*endstops)
+        steppers_being_homed = list()
+        for n in endstop_instances:
+            steppers_being_homed = steppers_being_homed + n.get_steppers()
+        steppernames_being_homed = list()
+        for n in steppers_being_homed:
+            steppernames_being_homed.append(n.get_name())
```

Comment:
FYI, you can do this in Python with: `steppernames = [s.get_name() for es, name in endstops for s in es.get_stepper()]`

Reply:
Told you in python I am an egg

Reply:
commit 2e2f7675390c9c6da1dbbaaf15575cd442e71827